packages/pancake-uikit/src/hooks/useTooltip/useTooltip.tsx

```diff
@@ -1,28 +1,36 @@
 import React, { useState, useEffect, useCallback, useRef } from "react";
-import { Placement, Padding } from "@popperjs/core";
 import { usePopper } from "react-popper";
+import { ThemeProvider, DefaultTheme } from "styled-components";
+import { light, dark } from "../../theme";
 import { StyledTooltip, Arrow } from "./StyledTooltip";
-import { TooltipRefs, TriggerType } from "./types";
+import { TooltipOptions, TooltipRefs } from "./types";
 
 function isTouchDevice() {
   return "ontouchstart" in window || navigator.maxTouchPoints > 0 || navigator.msMaxTouchPoints > 0;
 }
 
-const useTooltip = (
-  content: React.ReactNode,
-  placement: Placement = "auto",
-  trigger: TriggerType = "hover",
-  arrowPadding?: Padding,
-  tooltipPadding?: Padding,
-  tooltipOffset?: [number, number]
-): TooltipRefs => {
+const invertTheme = (currentTheme: DefaultTheme) => {
+  if (currentTheme.isDark) {
+    return light;
+  }
+  return dark;
+};
+
+const useTooltip = (content: React.ReactNode, options: TooltipOptions): TooltipRefs => {
+  const {
+    placement = "auto",
+    trigger = "hover",
+    arrowPadding = 16,
+    tooltipPadding = { left: 16, right: 16 },
+    tooltipOffset = [0, 10],
+  } = options;
   const [targetElement, setTargetElement] = useState<HTMLElement | null>(null);
   const [tooltipElement, setTooltipElement] = useState<HTMLElement | null>(null);
   const [arrowElement, setArrowElement] = useState<HTMLElement | null>(null);
 
   const [visible, setVisible] = useState(false);
-  const tooltipHoverRef = useRef(false);
-  const tooltipHoverTimeoutRef = useRef<number>();
+  const isHoveringOverTooltip = useRef(false);
+  const hideTimeout = useRef<number>();
```

Comment:
`useRef<typeof setTimeout>`

Reply:
yep, fixed. Can't wait when TS will start understanding if its browser or node projects 🙁

Reply:
Oh shit, CI pipeline didn't like the types.... ugh...

Reply:
OK, something is different between FE repo and Toolkit TS compilation. 
I'm aware of the return typeof setTimeout thing, used it myself [here](https://github.com/pancakeswap/pancake-frontend/blob/9cfe3a0e542da748a7b8e56f4789ccb223fffe75/src/hooks/useDelayedUnmount.ts#L17) and you used it [here](https://github.com/pancakeswap/pancake-frontend/blob/8a297c728af015a9e530648f1606453dbb634db4/src/hooks/useGetBlockCountdown.ts#L9) (btw I just noticed you use `typeof setTimeout` but you actually need `typeof setInterval`, although it doesn't really matter I guess).
In this repo TS is not ok with that during rollup build, even though it shows no problems in VSCode.

Anyway, I chose to specifically use browser's `window.setTimeout` that returns plain old `number` instead of complicated NodeJS `Timeout` type that is not applicable in browser anyway. It actually makes a bit more sense to me since this code is not supposed to be run outside of browser anyway, and in browser its `setTimeout` returns just number.